receiver/telemetrygeneratorreceiver/factory.go

```diff
@@ -0,0 +1,81 @@
+// Copyright observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package telemetrygeneratorreceiver //import "github.com/observiq/bindplane-agent/receiver/telemetrygeneratorreceiver"
+
+import (
+	"context"
+	"errors"
+
+	"github.com/observiq/bindplane-agent/receiver/telemetrygeneratorreceiver/internal/metadata"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/receiver"
+)
+
+// errImproperCfgType error for when an invalid config type is passed to receiver creation funcs
+var errImproperCfgType = errors.New("improper config type")
+
+// NewFactory creates a new receiver factory
+func NewFactory() receiver.Factory {
+	return receiver.NewFactory(
+		metadata.Type,
+		createDefaultConfig,
+		receiver.WithMetrics(createMetricsReceiver, metadata.MetricsStability),
+		receiver.WithLogs(createLogsReceiver, metadata.LogsStability),
+		receiver.WithTraces(createTracesReceiver, metadata.TracesStability),
+	)
+}
+
+// createDefaultConfig creates a default configuration
+func createDefaultConfig() component.Config {
+	return &Config{
+		PayloadsPerSecond: 1,
+		Generators: []*GeneratorConfig{
+			{
+				Type: GeneratorTypeLogs,
+			},
```

Comment:
There should be no generator for the default. A single generator is required on configuration so we don't want to supply one.